zerver/actions/message_edit.py

```diff
@@ -123,7 +124,10 @@ def validate_message_edit_payload(
     if propagate_mode != "change_one" and topic_name is None and stream_id is None:
         raise JsonableError(_("Invalid propagate_mode without topic edit"))
 
-    if message.realm.mandatory_topics and topic_name in ("(no topic)", ""):
+    if (
+        message.realm.topics_policy == RealmTopicsPolicyEnum.disable_empty_topic.value
+        and topic_name in ("(no topic)", "")
+    ):
         raise JsonableError(_("Topics are required in this organization."))
```

Comment:
@prakhar1144 while reviewing this, I noticed we still have an explicit `"(no topic)"` check here -- should that have been updated to have that topic name get coalesced into `""` before we get here?

Reply:
[#backend > general chat - migration discussion @ 💬](https://chat.zulip.org/#narrow/channel/3-backend/topic/general.20chat.20-.20migration.20discussion/near/2135869)